### PR TITLE
Report overlapping sections whenever possible

### LINKED ISSUE #284: More descriptive section link error messages
One quick example:
When the linker dies with error "Section FOO could not be loaded at $1337", it would be nice to see the name of the section that would overlap FOO (and this directly causes the error).

### PR DESCRIPTION
**Pull request by ISSOtm**
Fixes #284

### CODE REVIEW

File: src/link/assign.c [hunk 1]
```diff
@@ -284,32 +285,41 @@ static void placeSection(struct Section *section)
...
-			errx(1, "Unable to place \"%s\" (%s section) at $%02x:%04x",
-			     section->name, typeNames[section->type],
-			     section->bank, section->org);
+			snprintf(where, 64, "at $%02x:%04x",
+				 section->bank, section->org);
 		else if (section->isAlignFixed)
-			errx(1, "Unable to place \"%s\" (%s section) in bank %u with align mask %x",
-			     section->name, typeNames[section->type],
-			     section->bank, ~section->alignMask);
+			snprintf(where, 64, "in bank %02x with align mask %x",
```
Comment (aaaaaa123456789) [inline src/link/assign.c]: For consistency, the bank number should have a `$` before it

File: src/link/assign.c [hunk 2]
```diff
@@ -284,32 +285,41 @@ static void placeSection(struct Section *section)
...
+				 section->bank, ~section->alignMask);
 		else
-			errx(1, "Unable to place \"%s\" (%s section) in bank %u",
-			     section->name, typeNames[section->type],
-			     section->bank);
+			snprintf(where, 64, "in bank %02x", section->bank);
```
Comment (aaaaaa123456789) [inline src/link/assign.c]: Same here

**Comment by ISSOtm** [comment]
@aaaaaa123456789 Is that better?

**Comment by aaaaaa123456789** [comment]
Yeah, it seems OK now.

**Pull request merged**